server/repository/src/db_diesel/stock_line_ledger_discrepancy.rs

```diff
@@ -0,0 +1,82 @@
+use super::StorageConnection;
+
+use crate::diesel_macros::apply_equal_filter;
+use crate::{EqualFilter, RepositoryError, StockLineFilter};
+
+use crate::{db_diesel::stock_line_row::stock_line, DBType, StockLineRepository};
+
+use diesel::dsl::IntoBoxed;
+use diesel::prelude::*;
+
+table! {
+    stock_line_ledger_discrepancy (stock_line_id) {
+        stock_line_id -> Text,
+    }
+}
+
+joinable!(stock_line_ledger_discrepancy -> stock_line (stock_line_id));
+allow_tables_to_appear_in_same_query!(stock_line, stock_line_ledger_discrepancy);
+#[derive(Clone, Queryable, Debug, PartialEq)]
+#[diesel(table_name = stock_line_ledger_discrepancy)]
+pub struct StockLineLedgerDiscrepancy {
+    pub id: String,
+}
+
+pub struct StockLineLedgerDiscrepancyRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+pub struct StockLineLedgerDiscrepancyFilter {
+    pub stock_line: Option<StockLineFilter>,
+    pub stock_line_id: Option<EqualFilter<String>>,
+}
+
+impl<'a> StockLineLedgerDiscrepancyRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        StockLineLedgerDiscrepancyRepository { connection }
+    }
+
+    pub fn query(
+        &self,
+        filter: Option<StockLineLedgerDiscrepancyFilter>,
+    ) -> Result<Vec<StockLineLedgerDiscrepancy>, RepositoryError> {
+        let query = create_filtered_query(filter);
+
+        // Debug diesel query
+        // println!("{}", diesel::debug_query::<DBType, _>(&query).to_string());
+
+        let result =
+            query.load::<StockLineLedgerDiscrepancy>(self.connection.lock().connection())?;
+
+        Ok(result)
+    }
+}
+
+fn create_filtered_query(
+    filter: Option<StockLineLedgerDiscrepancyFilter>,
+) -> IntoBoxed<'static, stock_line_ledger_discrepancy::table, DBType> {
+    let mut query = stock_line_ledger_discrepancy::table.into_boxed();
+
+    let Some(StockLineLedgerDiscrepancyFilter {
+        stock_line,
```

Comment:
IDK if just me but I find the naming of this confusing. Without context I assume it's a `StockLine` struct at which point why is stock_line_id here separately. But it actually is a StockLineFilter... but we don't conventionally have struct field names in our filter structs that have the keyword "Filter" i.e. `StockLineLedgerDiscrepancyFilter::stock_line_filter: Option<StockLineFilter>`. Though I do think that'd be clearer... idk.

TBF I think it's clearer with IDE inlay hints, but you don't get that in github ;-). Maybe inlay hints make us not think through naming as much haha.

Reply:
We have had this pattern before, I'll make carry over to remove stock_line filter and just add store_id to StockLineLedgerDiscrepancy which is all we need atm tbh, thanks